src/wallet/rpcwallet.cpp

```diff
@@ -2575,6 +2577,261 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
+
+    // signature sizes can vary by a byte, so add 1 for each input when calculating the new fee
+    nNewFee = nNewFeeRate.GetFee(txSize + wtx.vin.size());
+
+    // if user set totalFee, use that instead
+    if (totalFee > 0) {
+        CAmount minTotalFee = nOldFeeRate.GetFee(txSize) + minRelayTxFee.GetFee(txSize);
+        if (totalFee < minTotalFee) {
+            strError = strprintf("Invalid totalFee, must be at least oldFee + relayFee: %s", FormatMoney(minTotalFee));
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strError);
+        }
+        nNewFee = totalFee;
+        nNewFeeRate = CFeeRate(totalFee, txSize);
+    }
+
+    // check that fee rate is higher than mempool's mininum fee
+    // (no point in bumping fee if we know that the new tx won't be accepted to the mempool)
+    // This may occur if the user set TotalFee or paytxfee too low, if fallbackfee is too low, or, perhaps,
+    // in a rare situation where the mempool minimum fee increased significantly since the fee estimation just a
+    // moment earlier. In this case, we report an error to the user, who may use totalFee to make an adjustment.
+    CFeeRate minMempoolFeeRate = mempool.GetMinFee(GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000);
+    if (nNewFeeRate.GetFeePerK() < minMempoolFeeRate.GetFeePerK()) {
+        strError = strprintf("New fee rate (%s) is too low to get into the mempool (min rate: %s)", FormatMoney(nNewFeeRate.GetFeePerK()), FormatMoney(minMempoolFeeRate.GetFeePerK()));
+        throw JSONRPCError(RPC_MISC_ERROR, strError);
+    }
+
+    CAmount nDelta = nNewFee - nOldFee;
+    if (nDelta <= 0) // it should not be possible to have a negative delta at this point (an attempt to reduce the fee)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "New fee must be higher than old fee");
+
+    {
+        LOCK(mempool.cs);
+        auto it = mempool.mapTx.find(hash);
+        if (it != mempool.mapTx.end()) {
+
+            // Tx with descendants
+            // --------------------
+            // Idea:  When a tx is bumped, its descendants (if any) are evicted from the mempool.
+            // Policy is that when you replace a tx, the total fees in the mempool cannot go down.
+            // So when you bump a tx that has children, you have to bump your fee by the sum of all of the
+            // descendants, plus the new bumped tx's relay fee.
+            //
+            // If the bumped fee is less than what's required because of children, we fail.  This is a different
+            // situation than bumping the fee to pay the minimum relay fee. Here, we can't be sure whether the user
+            // really wants to pay full price for all of the child transactions.  If so, the user can set payTxFee
+            // and run the command again.
+            //
+            CAmount nFeesWithDescendantsPlusRelay = it->GetModFeesWithDescendants() + ::minRelayTxFee.GetFee(txSize);
+            if (nNewFee < nFeesWithDescendantsPlusRelay) {
```

Comment:
I think this reduces bumpfee to a two step process in the common case (whenever there are children) and makes it so the second step is going to require setting totalfee.  It seems more user friendly to just pay what it takes?

On a separate note, I think it might be nice to check all of the children transactions and make sure their ancestor fee rate isn't higher than the new feerate you are bumping to.  If one of them has a higher ancestor fee rate you are actually make your situation worse.